python/cugraph-pyg/cugraph_pyg/examples/gcn_dist_sg.py

```diff
@@ -0,0 +1,178 @@
+
+wall_clock_start = time.perf_counter()
+device = torch.device("cuda")
+
+from ogb.nodeproppred import PygNodePropPredDataset  # noqa: E402
+
+dataset = PygNodePropPredDataset(name=args.dataset, root=args.dataset_root)
+split_idx = dataset.get_idx_split()
+data = dataset[0]
+
+graph_store = cugraph_pyg.data.GraphStore()
+graph_store[
+    ("node", "rel", "node"), "coo", False, (data.num_nodes, data.num_nodes)
+] = data.edge_index
+
+feature_store = cugraph_pyg.data.TensorDictFeatureStore()
+feature_store["node", "x"] = data.x
+feature_store["node", "y"] = data.y
+
+with tempfile.TemporaryDirectory(dir=args.tempdir_root) as samples_dir:
+    train_dir = os.path.join(samples_dir, "train")
+    os.mkdir(train_dir)
+    train_loader = NeighborLoader(
+        data=(feature_store, graph_store),
+        num_neighbors=[args.fan_out] * args.num_layers,
+        input_nodes=split_idx["train"],
+        replace=False,
+        batch_size=args.batch_size,
+        directory=train_dir,
+    )
+
+    val_dir = os.path.join(samples_dir, "val")
+    os.mkdir(val_dir)
+    val_loader = NeighborLoader(
+        data=(feature_store, graph_store),
+        num_neighbors=[args.fan_out] * args.num_layers,
+        input_nodes=split_idx["valid"],
+        replace=False,
+        batch_size=args.batch_size,
+        directory=val_dir,
+    )
+
+    test_dir = os.path.join(samples_dir, "test")
+    os.mkdir(test_dir)
+    test_loader = NeighborLoader(
+        data=(feature_store, graph_store),
+        num_neighbors=[args.fan_out] * args.num_layers,
+        input_nodes=split_idx["test"],
+        replace=False,
+        batch_size=args.batch_size,
+        directory=test_dir,
+    )
+
+    model = torch_geometric.nn.models.GCN(
+        dataset.num_features,
+        args.hidden_channels,
+        args.num_layers,
+        dataset.num_classes,
+    ).to(device)
+
+    optimizer = torch.optim.Adam(model.parameters(), lr=args.lr, weight_decay=0.0005)
+
+    warmup_steps = 20
+
+    def train(epoch: int):
+        model.train()
+        for i, batch in enumerate(train_loader):
+            if i == warmup_steps:
+                torch.cuda.synchronize()
+                start_avg_time = time.perf_counter()
+            batch = batch.to(device)
+
+            optimizer.zero_grad()
+            batch_size = batch.batch_size
+            out = model(batch.x, batch.edge_index)[:batch_size]
+            y = batch.y[:batch_size].view(-1).to(torch.long)
+
+            loss = F.cross_entropy(out, y)
+            loss.backward()
+            optimizer.step()
+
+            if i % 10 == 0:
+                print(f"Epoch: {epoch:02d}, Iteration: {i}, Loss: {loss:.4f}")
+        torch.cuda.synchronize()
+        print(
+            f"Average Training Iteration Time (s/iter): \
+                {(time.perf_counter() - start_avg_time)/(i-warmup_steps):.6f}"
+        )
+
+    @torch.no_grad()
+    def test(loader: NeighborLoader, val_steps: Optional[int] = None):
```

Comment:
Should move the function out of `with` block and just call it here.

Reply:
fixed